### PR TITLE
Dialogue edits and additions

### PR DESCRIPTION
**Pull request by Dweblenod**
A review of existing dialogues and the creation of new ones.

Some changes that are notable:
- The consort cult dialogue is more interactable and can result in the player gaining a Long Forgotten Warhorn
- Iguanas asking to eat your hair can now be given the ponytail item
- Dialogues which referred to family members have been made more ambiguous
- Dialogues where the response message was not the actual players words now use square brackets
- Some lore related dialogues have been updated to reflect other upcoming content

All dialogue is written by Dweblenod

### CODE REVIEW

**Comment by Dweblenod** [comment]
There is still some more dialogue I would consider adding and I will need to fix the merge conflicts but Im going to start marking this as ready for review

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 - Cruxite block set
 - New dialogue system that uses a gui screen, replacing the chat-based system
 - Dialogue in this new system is data-driven, and can be defined or tweaked through datapacks
+- New item tag for magic weapons
```
Comment (kirderf1) [inline CHANGELOG.md:24]: Not so sure that this is worth including in the changelog

**Review by kirderf1 — APPROVED** [review]
Great!

**Comment by Dweblenod** [comment]
Added one more commit, and will consider this done if the new commit passes review!

File: src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "condition": {
+          "type": "minestuck:list",
+          "conditions": [
+            {
+              "type": "minestuck:always_true"
+            }
+          ],
+          "list_type": "none"
+        },
+        "fail_tooltip": "No matter how you feel, you are compelled to decline",
```
Comment (kirderf1) [inline src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json:28]: Hang on, looks like there's a missing translation key here.

File: src/main/java/com/mraof/minestuck/data/dialogue/ConsortDialogue.java [hunk 3]
```diff
@@ -213,6 +213,37 @@ private static void consortDialogues(SelectableDialogueProvider provider, Dialog
...
 				new NodeBuilder(l.defaultKeyMsg("To eat or not to eat, that is the question.")));
+		provider.addRandomlySelectable("book_purpose", defaultWeight(all(isInTitleLand(THOUGHT), isAnyEntityType(NAKAGATOR))), new FolderedDialogue(builder ->
+		{
+			var buyMyProducts = builder.add("buy_my_products", new NodeBuilder()
+					.addMessage(l.subMsg("a", "I'm feeling generous, so how about this."))
+					.addMessage(l.subMsg("b", "If you want to get started on being an intellectual giant, I can offer you a discounted price on my self help memoir. It's only 5 payments of 8000 boondollars."))
+					.addDescription(l.subMsg("description", "They show you a book labelled \"Grindset Tales: From Pawn to King (How to follow the Philosopher's Journey)\""))
+					.addResponse(new ResponseBuilder(l.subMsg("purchase", "[Purchase book]")).visibleCondition(resource -> "No matter how you feel, you are compelled to decline", none(alwaysTrue())))
+					.addResponse(new ResponseBuilder(l.subMsg("decline", "Yeah no way.")))
+					.addResponse(new ResponseBuilder(l.subMsg("decline_hard", "Absolutely no chance.")))
```
Comment (kirderf1) [inline src/main/java/com/mraof/minestuck/data/dialogue/ConsortDialogue.java:224]: `addClosingResponse(msg)`?

File: src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json (hunk 2 above)
Comment (Dweblenod) [reply src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json:28]: Arnt normal fail tooltips untranslated?

Comment (kirderf1) [reply src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json:28]: The condition-default tooltips are untranslated, but they shouldn't be. For response-specific tooltip like this, you can use `l.subText()` in the same way that you use `l.subMsg()` (except that arguments aren't supported).

Comment (Dweblenod) [reply src/main/generated/resources/data/minestuck/minestuck/dialogue/consort/book_purpose/buy_my_products.json]: Alright, now done!

**Review by kirderf1 — APPROVED** [review]

**Comment by Dweblenod** [comment]
I see you self-requested a review, did you intend to review it more or should I take your approval as an opportunity to merge?

**Comment by kirderf1** [comment]
It's fine to merge by me.

**Pull request merged**